### PR TITLE
library: Add Progress Bar Entry

### PR DESCRIPTION
**Pull request by SoNiC-HeRE**
Added Progress Bar Entry under User Interface Section in Library
Issue #69

### CODE REVIEW

File: src/Library/demos/Progress Bar/main.blp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        margin-bottom: 24;
+      }
+
+      ProgressBar second {
+        inverted: true;
+        pulse-step: 0.25;
+        show-text: true;
+        text: "";
+        margin-bottom: 24;
+      }
```
Comment (andyholmes) [inline src/Library/demos/Progress Bar/main.blp:24]: I like the idea of having a second progress bar tracking the same value in a different way. I think the demonstration could be a little more intuitive, such as a "1/4 parts complete" or "time remaining" countdown.

File: src/Library/demos/Progress Bar/main.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const first_bar = workbench.builder.get_object("first");
+const second_bar = workbench.builder.get_object("second");
+const play = workbench.builder.get_object("play");
+
+function handleProgress() {
+  let counter = 0.3;
+  let counter_two = 0.25;
+
+  const intervalId_one = setInterval(() => {
```
Comment (andyholmes) [inline src/Library/demos/Progress Bar/main.js:9]: Add a brief comment above each of these intervals describing what they're doing, to help anyone skimming the source later. For example:

```js
// Advance the first progress bar 10% every 1.5 seconds
```

**Review by andyholmes — CHANGES_REQUESTED** [review]
Great idea, especially with the live demonstration!

File: src/Library/demos/Progress Bar/main.blp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      uri: "https://docs.gtk.org/gtk4/class.ProgressBar.html";
+      }
+
+    LinkButton{
+      label: "Human Interface Guidelines";
+      uri: "https://developer.gnome.org/hig/patterns/feedback/progress-bars.html";
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (SoNiC-HeRE) [reply src/Library/demos/Progress Bar/main.blp:43]: Sure, I'd fix it

File: src/Library/demos/Progress Bar/main.blp (hunk 1 above)
Comment (SoNiC-HeRE) [reply src/Library/demos/Progress Bar/main.blp:24]: Gotcha , I'll implement it

File: src/Library/demos/Progress Bar/main.js (hunk 2 above)
Comment (SoNiC-HeRE) [reply src/Library/demos/Progress Bar/main.js:9]: Sure

File: src/Library/demos/Progress Bar/main.blp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage {
+  title: "Progress Bar";
+  description: _("Display long running operation");
```
Comment (andyholmes) [inline src/Library/demos/Progress Bar/main.blp:6]:
```suggestion
  description: _("Display the progress of a long running operation");
```

**Review by andyholmes — APPROVED** [review]
Just this one change and it looks good to me, sorry I missed it last review.

**Pull request merged**